### ISSUE TITLE
[1.10.1] op_level_cost_estimator.cc:404] Check failed: 0 < gflops (0 vs. 0)

### ISSUE THREAD

**Issue by cdluminate** [tensorflow/tensorflow#22116]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: debian experimental
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: no
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: 1.10.1
- **Python version**: not related
- **Bazel version (if compiling from source)**: not related
- **GCC/Compiler version (if compiling from source)**: gcc8
- **CUDA/cuDNN version**: not related
- **GPU model and memory**: not related
- **Exact command to reproduce**:

### Describe the problem

I built tensorflow from source with a customized build system instead of bazel. https://salsa.debian.org/science-team/tensorflow . I can successfully build libtensorflow_cc.so according to the bazel query result. However when running `benchmark_model` test I encountered this problem:

```
$ LD_LIBRARY_PATH=. ./tf_benchmark_model --graph=../tensorflow_inception_graph.pb --num_threads=1
2018-09-06 15:45:23.316799: I tensorflow/tools/benchmark/benchmark_model.cc:469] Graph: [../tensorflow_inception_graph.pb.pb]
2018-09-06 15:45:23.316843: I tensorflow/tools/benchmark/benchmark_model.cc:470] Init ops:
2018-09-06 15:45:23.316848: I tensorflow/tools/benchmark/benchmark_model.cc:471] Input layers: [input:0]
2018-09-06 15:45:23.316852: I tensorflow/tools/benchmark/benchmark_model.cc:472] Input shapes: [1,224,224,3]
2018-09-06 15:45:23.316855: I tensorflow/tools/benchmark/benchmark_model.cc:473] Input types: [float]
2018-09-06 15:45:23.316859: I tensorflow/tools/benchmark/benchmark_model.cc:474] Output layers: [output:0]
2018-09-06 15:45:23.316862: I tensorflow/tools/benchmark/benchmark_model.cc:475] Target layers: []
2018-09-06 15:45:23.316868: I tensorflow/tools/benchmark/benchmark_model.cc:476] Num runs: [1000]
2018-09-06 15:45:23.316874: I tensorflow/tools/benchmark/benchmark_model.cc:477] Inter-inference delay (seconds): [-1.0]
2018-09-06 15:45:23.316898: I tensorflow/tools/benchmark/benchmark_model.cc:478] Inter-benchmark delay (seconds): [-1.0]
2018-09-06 15:45:23.316903: I tensorflow/tools/benchmark/benchmark_model.cc:480] Num threads: [1]
2018-09-06 15:45:23.316907: I tensorflow/tools/benchmark/benchmark_model.cc:481] Benchmark name: []
2018-09-06 15:45:23.316911: I tensorflow/tools/benchmark/benchmark_model.cc:482] Output prefix: []
2018-09-06 15:45:23.316915: I tensorflow/tools/benchmark/benchmark_model.cc:483] Show sizes: [0]
2018-09-06 15:45:23.316932: I tensorflow/tools/benchmark/benchmark_model.cc:484] Warmup runs: [1]
2018-09-06 15:45:23.316937: I tensorflow/tools/benchmark/benchmark_model.cc:251] Loading TensorFlow.
2018-09-06 15:45:23.316950: I tensorflow/tools/benchmark/benchmark_model.cc:258] Got config, 0 devices
2018-09-06 15:45:23.317031: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE3 SSE4.1 SSE4.2 AVX AVX2 FMA
2018-09-06 15:45:23.380132: I tensorflow/tools/benchmark/benchmark_model.cc:496] Initialized session in 0.063179s
2018-09-06 15:45:23.380206: I tensorflow/tools/benchmark/benchmark_model.cc:327] Running benchmark for max 1 iterations, max -1 seconds without detailed stat logging, with -1s sleep between inferences
2018-09-06 15:45:23.516527: F tensorflow/core/grappler/costs/op_level_cost_estimator.cc:404] Check failed: 0 < gflops (0 vs. 0)type: "CPU"
vendor: "GenuineIntel"
model: "110"
num_cores: 4
environment {
  key: "cpu_instruction_set"
  value: "SSE, SSE2"
}
environment {
  key: "eigen"
  value: "3.3.90"
}
l1_cache_size: 32768
l2_cache_size: 262144
l3_cache_size: 6291456
memory_size: 8558383104

Aborted
```

the `tensorflow_inception_graph.pb` files comes from `tensorflow/contrib/makefile/README.md`

### Source code / logs

https://salsa.debian.org/science-team/tensorflow

For error log see above.

**Comment by cdluminate** [comment]
The cause of this problem is that `device.frequency()` returned `0`.
```c++
t/c/g/c/op_level_cost_estimator.cc                                 [buffers] 
 363 
 364   if (device.type() == "CPU") {
 365     // Check if vector instructions are available, and refine performance
 366     // prediction based on this.
 367     // Frequencies are stored in MHz in the DeviceProperties.
 368     gflops = device.num_cores() * device.frequency() * 1e-3;
 369     if (gb_per_sec < 0) {
 370       if (device.bandwidth() > 0) {
 371         gb_per_sec = device.bandwidth() / 1e6;
 372       } else {
 373         gb_per_sec = 32;
 374       }
 375     }
```

**Comment by tensorflowbutler** [comment]
Nagging Assignee @cy89: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by victorbocharov** [comment]
It seems that I have a similar problem.
`device.frequency()` returned `0` probably because `NominalCPUFrequency()` returned `1`. `TENSORFLOW_USE_ABSL` turns on abseil support and it is used only in Bazel-based compilation.

https://github.com/tensorflow/tensorflow/blob/v1.10.1/tensorflow/core/platform/posix/port.cc#L196 :
```
double NominalCPUFrequency() {
#ifdef TENSORFLOW_USE_ABSL
  return absl::base_internal::NominalCPUFrequency();
#else
  return 1.0;
#endif
}
```

Fixed here for Makefile build but not for cmake build: https://github.com/tensorflow/tensorflow/commit/75adfa5ca180abdd1ccf9408078906e5874c9e2a

**Comment by tensorflowbutler** [comment]
Nagging Assignee @cy89: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @cy89: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @cy89: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @cy89: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by ankitadhar** [comment]
I have similar issue..
I am getting following error:

`
I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:116] None of the MLIR optimization passes are enabled (registered 2)`
`W tensorflow/core/platform/profile_utils/cpu_utils.cc:126] Failed to get CPU frequency: 0 Hz`
`F tensorflow/core/grappler/costs/op_level_cost_estimator.cc:710] Check failed: 0 < gflops (0 vs. 0)type: "CPU"
`

I am using mac with M1 chip. I don't understand these errors much. Is there any way to fix it from my end?

**Comment by ParseDark** [comment]
Same issues. How is going on guys?

**Comment by ParseDark** [comment]
> Same issues. How is going on guys?
> 
> I also use mac with M1 chip.

Just need to run it:
```
export CUDA_VISIBLE_DEVICES=""
```

**Comment by ankitadhar** [comment]
Have a look at https://stackoverflow.com/questions/67379519/cannot-use-keras-models-on-mac-m1-with-bigsur

**Comment by ITvampire** [comment]
> > Same issues. How is going on guys?
> > I also use mac with M1 chip.
> 
> Just need to run it:
> 
> ```
> export CUDA_VISIBLE_DEVICES=""
> ```

Could you talk about where to run it?

**Comment by mohantym** [comment]
Have you checked this [thread ](https://developer.apple.com/metal/tensorflow-plugin/)on install Tensorflow in macos?
